include/cbl/CBLDatabase.h

```diff
@@ -54,11 +54,10 @@ typedef struct CBLEncryptionKey {
 
 /** Database configuration options. */
 typedef struct {
-    FLString directory;                     ///< The parent directory of the database
-    CBLEncryptionKey* encryptionKey;        ///< The database's encryption key (if any)
+    FLString directory;                         ///< The parent directory of the database
+    const CBLEncryptionKey* encryptionKey;      ///< The database's encryption key (if any)
```

Comment:
This is also ifdef on other platforms

Reply:
OK, I have ifdef the encryptionKey and related. There are some implementation that needs to be branch accordingly as well and it's not very pretty. Please review again.